### PR TITLE
Alt. 1: Duplicated IpAddress and Port variables for both protocols

### PR DESCRIPTION
**Pull request by PTaeuberDS**
This is the same PR as #27 but without one [commit](https://github.com/modelica/fmi-ls-xcp/pull/27/commits/ae24da66111cd36babdef368d4b194c073e45836) that is added to #27 and adds another configuration parameter for the protocol.
In this PR (#26) the IpAddress and port variables are duplicated for the different protocol types for the special case that an XCP service uses TCP and UDP channels in parallel.

### CODE REVIEW

**Comment by PTaeuberDS** [comment]
@andreas-junghanns @pmai 
@ everybody interested

The PR is ready for review. Due to the open discussion in #19 there are two variants of this PR now (see PR description).

File: docs/index.adoc [hunk 1]
```diff
@@ -361,18 +420,18 @@ In the latter case, the importer or an external A2L tool is responsible for dete
...
 In this case the FMU importer must provide its own implementation of an XCP slave.
 
-It is also possible that the FMU does contain an internal XCP slave implementation but additionally allows the importer to access the memory of the virtual ECU with an external XCP service (attribute `supportsExternalXcpService = true`).
+It is also possible that the FMU does contain an internal XCP slave implementation but additionally allows the importer to access the memory of the virtual ECU with an external XCP service (attribute `supportsDirectMemoryAccessViaA2L = true`).
 In this case it is the choice of the importer whether to use the FMU internal XCP slave or its own implementation.
 See <<starting-internal-xcp-service>> on how to deactivate the internal XCP service.
 
-_[Note that the FMU must set at least one of the attributes `containsXcpService` or `supportsExternalXcpService` to `true` to make use of this layered standard.]_
+_[Note that the FMU must set at least one of the attributes `containsXcpService` or `supportsDirectMemoryAccessViaA2L` to `true` to make use of this layered standard.]_
```
Comment (andreas-junghanns) [inline docs/index.adoc:434]: Should we mandate this? Now, this is at best a non-normative recommendation. We should consider being stronger here.

**Review by andreas-junghanns — APPROVED** [review]
Thank you: great work!

File: docs/index.adoc (hunk 1 above)
Comment (PTaeuberDS) [reply docs/index.adoc:437]: I will do this in a following PR (tracked in Issue #15).

**Comment by PTaeuberDS** [comment]
@andreas-junghanns I merged your review PR from main into this PR. Can we also merge this one and make this the new baseline? Then we can decide with the other PR if we want to go for the other alternative afterwards.

**Review by andreas-junghanns — APPROVED** [review]
Big step forward. I think "FMI 2" and "FMI 3" should be replaced with "FMI 2.0" and "FMI 3.0" as this is what we use elsewhere. I know you mean FMI 2.X, which is technically more correct, but it looks strange if this is the only FMI standard text (is it?) that uses these. Can be fixed later.

**Pull request merged**